driver/main.ml

```diff
@@ -59,10 +59,7 @@ module Options = Main_args.Make_bytecomp_options (struct
     | None -> () (* this should not occur as we use Arg.Symbol *)
     | Some pass ->
         stop_after := Some pass;
-        begin match pass with
-        | P.Parsing | P.Typing ->
-            compile_only := true
-        end;
+        compile_only := P.is_compilation_pass pass
```

Comment:
This change (and the one in optmain) needs to be moved to `Main_args.Default.Compiler`, sorry about that.

Reply:
Rebased, fixed, and updated .depend (but the change to .depend seems unrelated to this PR).